From a4212e47ff20b415e042682944a82623b3f75cee Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 29 May 2019 10:21:32 -0400
Subject: [PATCH 23/63] Do a better job of making sure -DLIBEFIVAR_VERSION
 happens...

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 Make.defaults | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Make.defaults b/Make.defaults
index 6c575846687..80fd99eba71 100644
--- a/Make.defaults
+++ b/Make.defaults
@@ -18,7 +18,7 @@ CCLD_FOR_BUILD	?= $(CC_FOR_BUILD)
 CCLD	:= $(if $(filter undefined,$(origin CCLD)),$(CC),$(CCLD))
 OPTIMIZE ?= -O2 -flto
 CFLAGS	?= $(OPTIMIZE) -g3
-CFLAGS	:= $(CFLAGS) -DLIBEFIVAR_VERSION=$(VERSION)
+CFLAGS	:= $(CFLAGS)
 LDFLAGS ?=
 LDFLAGS := $(LDFLAGS)
 AR	:= $(CROSS_COMPILE)$(COMPILER)-ar
@@ -29,6 +29,8 @@ ABIDIFF := abidiff
 
 PKGS	=
 
+CPPFLAGS += -DLIBEFIVAR_VERSION=$(VERSION)
+
 clang_cflags = -D_GNU_SOURCE -std=gnu11 -Wno-address-of-packed-member
 gcc_cflags = -specs=$(TOPDIR)/gcc.specs
 cflags	= $(CFLAGS) -I${TOPDIR}/src/include/ \
-- 
2.26.2

