From 7cfb9bd539ced0fd6fe35e7c08326bc3011d059b Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 18 Jun 2019 11:06:09 -0400
Subject: [PATCH 28/63] Make: Add some more stuff to the toplevel 'clean'

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8c67b9d2654..9c2620cb485 100644
--- a/Makefile
+++ b/Makefile
@@ -49,7 +49,7 @@ clean :
 	@set -e ; for x in $(SUBDIRS) ; do \
 		$(MAKE) -C $$x $@ ; \
 	done
-	@rm -vf efivar.spec
+	@rm -vf efivar.spec vgcore.* core.*
 
 test-archive: abicheck efivar.spec
 	@rm -rf /tmp/efivar-$(GITTAG) /tmp/efivar-$(GITTAG)-tmp
-- 
2.26.2

