From 20928bf97924328d41010aecdbf925c37f633739 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 19 Jun 2019 11:35:28 -0400
Subject: [PATCH 29/63] Make scan build rules slightly more intuitive.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 Make.scan-build | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/Make.scan-build b/Make.scan-build
index 7d684f7539f..f508e30be16 100644
--- a/Make.scan-build
+++ b/Make.scan-build
@@ -12,10 +12,9 @@ scan-build : | scan-test
 scan-build : clean
 	scan-build -o scan-results make $(DASHJ) CC=clang all
 
-scan-build-all : | scan-test
-scan-build-all : clean
-	scan-build -o scan-results make $(DASHJ) CC=clang all
+scan-build-all: | scan-build
+scan : | scan-build
 
-.PHONY : scan-build scan-clean
+.PHONY : scan-build scan-clean scan-build-all scan
 
 # vim:ft=make
-- 
2.26.2

