From 150873b00fdb2d5aca2e7c66d0545ad55801409a Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Fri, 11 Oct 2019 10:44:53 -0400
Subject: [PATCH 44/63] Fix some 32-bit size_t format specifier errors

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/export.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/export.c b/src/export.c
index cfb021525ff..732b4bf30ad 100644
--- a/src/export.c
+++ b/src/export.c
@@ -149,7 +149,7 @@ overflow:
 	}
 
 	crc = efi_crc32(data, size - sizeof(uint32_t));
-	debug("efi_crc32(%p, %lu) -> 0x%"PRIx32", expected 0x%"PRIx32,
+	debug("efi_crc32(%p, %zu) -> 0x%"PRIx32", expected 0x%"PRIx32,
 	      data, size - sizeof(uint32_t), crc,
 	      *(uint32_t*)(data + size - sizeof(uint32_t)));
 
@@ -282,7 +282,7 @@ efi_variable_import_efivar(uint8_t *data, size_t datasz, efi_variable_t **var_ou
 		}
 
 		crc = efi_crc32(data, datasz - sizeof(uint32_t));
-		debug("efi_crc32(%p, %lu) -> 0x%"PRIx32", expected 0x%"PRIx32,
+		debug("efi_crc32(%p, %zu) -> 0x%"PRIx32", expected 0x%"PRIx32,
 		      data, datasz - sizeof(uint32_t), crc,
 		      *(uint32_t*)(data + datasz - sizeof(uint32_t)));
 
@@ -473,7 +473,7 @@ overflow:
 	ptr += var->data_size;
 
 	crc = efi_crc32(data, needed - sizeof(uint32_t));
-	debug("efi_crc32(%p, %lu) -> 0x%"PRIx32,
+	debug("efi_crc32(%p, %zu) -> 0x%"PRIx32,
 	      data, needed - sizeof(uint32_t), crc);
 	*(uint32_t *)ptr = crc;
 
@@ -611,7 +611,7 @@ overflow:
 	ptr += var->data_size;
 
 	crc = efi_crc32(data, needed - sizeof(uint32_t));
-	debug("efi_crc32(%p, %lu) -> 0x%"PRIx32,
+	debug("efi_crc32(%p, %zu) -> 0x%"PRIx32,
 	      data, needed - sizeof(uint32_t), crc);
 	*(uint32_t *)ptr = crc;
 
-- 
2.26.2

