From 45d1aa955a38541ba8e1554026847e582ae28bfd Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 16 Oct 2019 10:01:00 -0400
Subject: [PATCH 53/63] gitignore: ignore *.strace

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 947d88eec42..f61b0bcb340 100644
--- a/.gitignore
+++ b/.gitignore
@@ -14,6 +14,7 @@
 *.so
 *.so.*
 *.spec
+*.strace
 *.tar.*
 *.var
 core.*
-- 
2.26.2

